### PR TITLE
Private/mmeeks/queuecleanup2

### PR DESCRIPTION
**Pull request by mmeeks**
* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: kit/Kit.cpp [hunk 1]
```diff
@@ -2681,7 +2682,9 @@ int KitSocketPoll::kitPoll(int timeoutMicroS)
         do
         {
             int realTimeout = timeoutMicroS;
-            if (_document && _document->hasQueueItems() && _document->processInputEnabled())
+            if (_document &&
+                (_document->hasCallbacks() ||
+                 (_document->hasQueueItems() && _document->processInputEnabled())))
```
Comment (caolanm) [inline kit/Kit.cpp:2687]: Reassure me that this  (a || b && c) should not be (a || b) && c, we want this as it is written?

File: kit/KitQueue.hpp [hunk 2]
```diff
@@ -42,11 +42,53 @@ class KitQueue
...
+        std::string _payload;
+
+        Callback() : _view(-1), _type(-1) { }
+        Callback(int view, int type, const std::string payload) :
+            _view(view), _type(type), _payload(payload) { }
+
+        static std::string toString(int view, int type, const std::string payload)
+        {
+            std::ostringstream str;
+            str << view << " " << lokCallbackTypeToString(type) << " "
```
Comment (caolanm) [inline kit/KitQueue.hpp:57]: iOS build seems to suggest that #include <LibreOfficeKit/LibreOfficeKitEnums.h> is newly needed to build

File: kit/Kit.cpp (hunk 1 above)
Comment (mmeeks) [reply kit/Kit.cpp:2687]: Yes - it is correct; I can split it into two happier named booleans if that helps =)

Comment (mmeeks) [reply kit/Kit.cpp]: Split to its own function & commented it :-)

Comment (caolanm) [reply kit/Kit.cpp]: great :-)

**Review by caolanm — APPROVED** [review]

**Pull request merged**